src/include/kernel/ext_manifest.h

```diff
@@ -73,6 +73,20 @@ enum ext_man_elem_type {
 	EXT_MAN_ELEM_FW_VERSION		= 0,
 	EXT_MAN_ELEM_CC_VERSION		= SOF_IPC_EXT_CC_INFO,
 	EXT_MAN_ELEM_PROBE_INFO		= SOF_IPC_EXT_PROBE_INFO,
+	EXT_MAN_ELEM_CONFIG_DATA,
+};
+
+/* EXT_MAN_ELEM_CONFIG_DATA elements */
+#define EXT_MAN_ELEM_CONFIG_NUM	1
```

Comment:
This cant be fixed size otherwise it's an ABI change. The number of elems needs to be variable size.

Reply:
This is local variable, kernel will still use only .hdr.elem_size

Reply:
ok, but we should use ARRAY_SIZE() instead of this.

Reply:
hmm, we cant use ARRAY_SIZE() with enum. What about moving this define to enum list and there will be something like EXT_MAN_CONFIG_LAST_ELEM always at the end of enum?

Reply:
@lgirdwood please check now